### PR TITLE
chore(localdebug): add new prerequisite task

### PR DESCRIPTION
**Pull request by XiaofuHuang**
Add new prerequisite check task
Detail: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/15310381

### CODE REVIEW

File: packages/vscode-extension/package.json [hunk 1]
```diff
@@ -736,7 +736,81 @@
         ],
         "properties": {
           "command": {
-            "type": "string"
+            "oneOf": [
```
Comment (swatDong) [inline packages/vscode-extension/package.json:739]: Does this keep backward compatibility, e.g., `teamsfx: ngrok start`?

File: packages/vscode-extension/package.json [hunk 2]
```diff
@@ -736,7 +736,81 @@
...
+              {
+                "const": "debug-check-prerequisites",
+                "description": "Validate and install prerequisites.\n See https://aka.ms/teamsfx-debug-tasks#check-prerequisites for how to customize following args."
```
Comment (swatDong) [inline packages/vscode-extension/package.json:742]: Any localization support?

File: packages/vscode-extension/src/debug/prerequisitesHandler.ts [hunk 3]
```diff
@@ -336,12 +337,37 @@ export async function checkAndInstall(): Promise<Result<void, FxError>> {
   );
 }
 
-// TODO: use _checkAndInstall for new prerequisite task
-// export async function checkAndInstallV2(
-//   orderedCheckers: PrerequisiteOrderedChecker[],
-//   ports: number[]
-// ): Promise<Result<void, FxError>> {
-// }
+export async function checkAndInstallV2(
```
Comment (swatDong) [inline packages/vscode-extension/src/debug/prerequisitesHandler.ts:340]: Better naming than v2...
- `checkAndInstallForTask`
- or, `checkAndInstallHandler`
- or, renaming the old one to `checkAndInstallLegacy`

Comment (XiaofuHuang) [reply packages/vscode-extension/src/debug/prerequisitesHandler.ts]: Updated it to `checkAndInstallForTask`, thx.

File: packages/vscode-extension/package.json (hunk 2 above)
Comment (XiaofuHuang) [reply packages/vscode-extension/package.json]: Use the string in `package.nls.json` instead.

File: packages/vscode-extension/package.json (hunk 1 above)
Comment (XiaofuHuang) [reply packages/vscode-extension/package.json]: As we discussed offline, command can be anything. The auto prompt only contains `debug-check-prerequisites`. If the command is invalid, the error message will be shown after execution.

File: packages/vscode-extension/src/debug/taskTerminal/prerequisiteTaskTerminal.ts [hunk 4]
```diff
@@ -3,13 +3,24 @@
...
 
+export enum Prerequisite {
+  nodejs = "nodejs",
+  m365Account = "m365Account",
+  devCert = "devCert",
+  func = "func",
+  ngrok = "ngrok",
+  dotnet = "dotnet",
+  ports = "ports",
+}
```
Comment (kuojianlu) [inline packages/vscode-extension/src/debug/taskTerminal/prerequisiteTaskTerminal.ts:22]: Will `PrerequisiteArgs` and `Prerequisite` be used in CLI in the future?

File: packages/vscode-extension/src/debug/prerequisitesHandler.ts [hunk 5]
```diff
@@ -1148,6 +1174,46 @@ async function getOrderedCheckersForGetStarted(): Promise<PrerequisiteOrderedChe
   }
 }
 
+async function getOrderedCheckersV2(
```
Comment (kuojianlu) [inline packages/vscode-extension/src/debug/prerequisitesHandler.ts:1177]: Rename this also?

Comment (swatDong) [inline packages/vscode-extension/src/debug/prerequisitesHandler.ts:1177]: Another `V2`~

File: packages/vscode-extension/src/debug/taskTerminal/prerequisiteTaskTerminal.ts (hunk 4 above)
Comment (XiaofuHuang) [reply packages/vscode-extension/src/debug/taskTerminal/prerequisiteTaskTerminal.ts:22]: `PrerequisiteArgs` may be different in CLI, `prerequisite` may be same.
What about move them into core after CLI design is ready?

File: packages/vscode-extension/src/debug/prerequisitesHandler.ts (hunk 5 above)
Comment (XiaofuHuang) [reply packages/vscode-extension/src/debug/prerequisitesHandler.ts]: Sorry, renamed it.

**Review by swatDong — APPROVED** [review]

File: packages/vscode-extension/src/debug/taskTerminal/prerequisiteTaskTerminal.ts (hunk 4 above)
Comment (kuojianlu) [reply packages/vscode-extension/src/debug/taskTerminal/prerequisiteTaskTerminal.ts:22]: Sure.

**Review by kuojianlu — APPROVED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Pull request merged**